examples/string_token_substitution.ipynb

```diff
@@ -0,0 +1,247 @@
+{
+ "cells": [
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "# String token substitution",
+   "id": "4c4a5a5ed39eb2e1"
+  },
+  {
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-06-05T09:33:38.946449Z",
+     "start_time": "2024-06-05T09:33:38.943885Z"
+    }
+   },
+   "cell_type": "code",
+   "source": "from resources import helpers",
+   "id": "8b19d0e7242a0a53",
+   "outputs": [],
+   "execution_count": 7
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "When a host application `resolve`s a string from the manager, it might be that the string contains placeholder tokens, which must be substituted with values held by the host application before the string can be used. Similarly, the host may publish an entity with a string property that contains placeholder tokens, which the manager must substitute before the string can be used.\n",
+    "\n",
+    "Given that, typically, the host application and manager plugin are independently developed, this presents a few problems\n",
+    "\n",
+    "* Hosts/managers need a way to find out if a string needs substitution at all.\n",
+    "* Hosts/managers need a shared understanding of how placeholder tokens are encoded within a string.\n",
+    "* Hosts/managers need to know which tokens are possible/allowed within a given string.\n",
+    "* Hosts/managers need the ability to substitute values into the placeholder tokens in a string.\n",
+    "\n",
+    "OpenAssetIO solves these problems by a combination of documentation and a handy `substitute` utility function. The documentation of individual traits and their properties will tell hosts/managers whether they can place substitution tokens in a string, what tokens are allowed, and what values to map to each token. A common OpenAssetIO-dictated format for substitution tokens ensures that hosts/managers know how placeholders should be presented within strings. Once the host/manager has gathered the necessary values, they can use the `substitute` utility function to perform the substitution."
+   ],
+   "id": "feec93d227a1e95e"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "\n",
+    "## Example - LocatableContent\n",
+    "\n",
+    "Let's take an example from the MediaCreation library. The LocatableContent trait has an `\"isTemplated\"` property, which, if true, indicates that the `\"location\"` property contains substitution tokens.\n",
```

Comment:
[consider]
This reads a little like `isTemplated` is a `LocatableContent` specific property (which I guess it _sort of is_ right now, but I don't think that's how we should frame it.

Perhaps?

```suggestion
    "Let's take an example from the MediaCreation library. 
    
    Before anything else, a host needs to know if they need to account for substitutions. The LocatableContent trait has an `\"isTemplated\"` property, which, if true, indicates that the `\"location\"` property contains substitution tokens.\n",
```

Reply:
Added verbatim in c80901c9